### DISCUSSION TITLE
Cannot get user name from AppleProvider

### DISCUSSION THREAD

**Discussion by CarstenHoyer** [nextauthjs/next-auth#7560] [Help]
Hi, I use the standard AppleProvider setup. I can sign in and I do see my email in the response, but not my name. I have tried updating the authorization params, but even if I just ask for the name, it does not come back from the response.

I go into https://appleid.apple.com/account/manage and stop using "Sign in with Apple" on the relevant app - because i have read that Apple only returns the name on the first sign in.

How can I get the name in my sign in response?

**Comment by CarstenHoyer** [comment]
This is my AppleProvider:

```
AppleProvider({
      clientId: process.env.APPLE_CLIENT_ID,
      clientSecret: appleJWT(),
      authorization: {
        params: { scope: "name email", response_mode: "form_post" },
      },
      async profile(profile) {
        console.log({ profile })

        return {
          ....
        }
      }
    }),
```
    

This is the profile I see in the above console.log:


```
{
  profile: {
    iss: 'https://appleid.apple.com',
    aud: '...',
    exp: 1684311661,
    iat: 1684225261,
    sub: '...',
    at_hash: '...',
    email: '...',
    email_verified: 'true',
    auth_time: 1684225260,
    nonce_supported: true
  }
}

```

**Comment by orendecor** [comment]
I'm having the same problem

**Comment by BuhayovA** [comment]
The same issue :(
